pynetbox_query/pynetboxquery/utils/read_data.py

```diff
@@ -0,0 +1,96 @@
+from pathlib import Path
+from typing import List, Dict
+from pynetboxquery.utils.device_dataclass import Device
+from pynetboxquery.utils.error_classes import (
+    FileTypeNotSupported,
+    DelimiterNotSpecifiedError,
+    SheetNameNotSpecifiedError,
+)
+from pynetboxquery.utils.read_methods import _ReadMethods
+
+
+# Disabling this pylint warning as it is unnecessary.
+# pylint: disable = R0903
+class ReadData(_ReadMethods):
+    """
+    This class contains methods to read data from different file types into a list of dictionaries.
+    Only the "read" method should be called by the user as it can differentiate between different file types.
+    The rest of the methods read data from the file into a list of dictionaries.
+    """
+
+    def read(self, file_path: str, **kwargs) -> List[Device]:
```

Comment:
incidentally - this removes the need for `validate_kwargs`/ `check_filepath`  functions. Since that should be handled by the specific `Read` method

i.e. 

```
class ReadCSV(ReadFile):
    def __init__(**kwargs):
        self._validate(kwargs)
    
    def _validate(params: Dict):
        if "sheet_name" not in kwargs:
            raise error
    
    def read(file_path: str):
        self._check_file_path(file_path) # inherited from ReadFile Abstract Class
        with open(file_path, mode="r", encoding="UTF-8") as file:
            csv_reader = DictReader(file)
            dictionary_list = list(csv_reader)
        return dictionary_list